src/vm/virtualMachine.ts

```diff
@@ -28,20 +25,113 @@ export class VirtualMachine {
     );
 
     if (maybeEncodedInstruction === undefined) {
-      throw new MemoryError(EndOfInstructionsError);
+      throw new BaseError(ErrorType.MemoryError, EndOfInstructionsError);
     }
 
     const encodedInstruction = maybeEncodedInstruction;
 
     if (!(encodedInstruction instanceof Felt)) {
-      throw new MemoryError(InstructionError);
+      throw new BaseError(ErrorType.MemoryError, InstructionError);
     }
 
-    const maybeUint = encodedInstruction.toUint64();
+    const { value: encodedInstructionUint, error } =
+      encodedInstruction.toUint64();
+    if (error !== undefined) {
+      throw error;
+    }
     // decode and run instruction
+    const instruction = Instruction.decodeInstruction(encodedInstructionUint);
+
     // return this.runInstruction();
     throw new Error('TODO: Not Implemented');
   }
 
-  computeOperands(instruction: Instruction) {}
+  // Compute the operands of an instruction. The VM can either
+  // fetch them from memory and if it fails to do so, it can
+  // deduce them from the instruction itself.
+  computeOperands(instruction: Instruction): Result<void> {
```

Comment:
why does it return void?
is it a todo?

Reply:
yes, currently working on the rest of the computation that will return the operands